syntax = "proto3";

package mafia;

service Mafia {
    rpc GoSingUp(SingUp) returns (SingUpResponse) {}
    rpc SubscribeToNotifications (SubscribeRequest) returns (stream SubscribeResponse) {}
    rpc DisconectRoom(SingUp) returns (Empty) {}
    rpc ConnectRoom(SingUp) returns (SingUpResponse) {}
    rpc DeadSignal(SingUp) returns (Empty) {}
    rpc CityVoting(CityVotingRequest) returns (CityVotingResponse) {}
    rpc KillCitizen(KillCitizenRequest) returns (Empty) {}
    rpc CheckCitizen(CheckCitizenRequest) returns (Empty) {}
    rpc GetNightResult(GetNightResultRequest) returns (GetNightResultResponse) {}
    rpc GetRole(SingUp) returns (GetRoleResponse) {}
}

message GetRoleResponse {
    string role = 1;
}

message CityVotingRequest {
    string game_id = 1;
    string name = 2;
    string vote = 3;
}

message CityVotingResponse {
    bool is_end = 1;
    string end = 2;
    repeated string city = 3;
}

message KillCitizenRequest {
    string game_id = 1;
    string name = 2;
}

message CheckCitizenRequest {
    string game_id = 1;
    string name = 2;
}

message CheckCitizenResponse {
    bool result = 1;
}

message GetNightResultRequest {
    string name = 1;
    string game_id = 2;
}

message GetNightResultResponse {
    bool is_end = 1;
    string end = 2;
    repeated string city = 3;
}

message SingUp {
    string name = 1;
    string game_id = 2;
}

message SingUpResponse {
    string game_id = 1;
    repeated string players = 2;
}

message SubscribeRequest {
    string name = 1;
    string game_id = 2;
}

message SubscribeResponse {
    string msg = 1;
}

message Empty {}
